samples/Mvc.ControllerForMultipleAreasSample/MultipleAreasControllerConvention.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+namespace ControllerForMultipleAreasSample
+{
+    public class MultipleAreasControllerConvention : IApplicationModelConvention
```

Comment:
You can also use `IControllerModelConvention` instead and then you don't need the loop.

Reply:
How will the new copied controllers with different area value in its RouteValues be added to the application?

Reply:
Not explained here but we ran into a problem with this approach. You can't modify `application.Controllers` while inside an `IControllerModelConvention`.